tools/unstableWarningAnonymizer/unstableAnonScript.chpl

```diff
@@ -31,84 +31,187 @@ features.
+  // Check if the string mentions that something is unstable
+  return warning.find("unstable") != -1 || warning.find("enum-to-bool") != -1 ||
+         warning.find("enum-to-float") != -1;
+}
+
 
-inline proc prettyPrintMap(m : map(?), writer: IO.fileWriter(?)){
-  for key in m.keys() {
-    if m[key] < 2 then
-      writer.writeln(m[key], " instance of \"", key, "\"");
-    else writer.writeln(m[key], " instances of \"", key, "\"");
+proc anonmizeWarning(warning: string) : string {
+  // Anonymize known warning messages that include variable names
+  // when so that it doens't reveal variable names or other impl detauls
+
+  const forallRef ="warning: inferring a 'ref' intent on an array in a forall is unstable - in the future this may require an explicit 'ref' forall intent for";
+  if warning.find(forallRef) != -1 then
+    return forallRef + " <var-name>";
+
+  const typeName = "warning: using a type's name ";
+  const typeNameUse = "in a 'use' statement to access its tertiary methods is an unstable feature";
+  const typeNameImport = "in an 'import' statement to access its tertiary methods is an unstable feature";
+  if warning.find(typeName) != -1 && warning.find(typeNameUse) != -1 then
+    return typeName + typeNameUse;
+  if warning.find(typeName) != -1 && warning.find(typeNameImport) != -1 then
+    return typeName + typeNameImport;
+
+  const underscore = "warning: symbol names with leading underscores";
+  const end = " are unstable";
+  if warning.find(underscore) != -1 && warning.find(end) != -1 then
+    return underscore + end;
+
+  const chpl_ = "warning: symbol names beginning with 'chpl_' ";
+  if warning.find(chpl_) != -1 && warning.find(end) != -1 then
+    return chpl_ + end;
+
+  const pragmas = "uses pragmas, which are considered unstable and may change in the future";
+  if warning.find(pragmas) != -1 then
+    return "warning: <proc> " + pragmas;
+
+  return warning;
+}
+
+inline proc prettyPrintArr(arr : [] (string, int, int), writer: IO.fileWriter(?), fileCount: bool){
+  for a in arr {
+    const grammar = if a[1] < 2 then " instance of \"" else " instances of \"";
+    const files = if fileCount then
+                  "\" across " + a[2] :string + if a[2] < 2 then
+                  " file" else " files"
+                  else "\"";
+    writer.writeln(a[1], grammar, a[0], files);
+  }
+}
+
+
+inline proc csvPrintArr(arr : [] (string, int, int), writer: IO.fileWriter(?), fileCount: bool){
+  writer.writeln("warning", ",", "count", if fileCount then ",uniqueFiles" else "");
+  for a in arr {
+    const files = if fileCount then
+                  "," + a[2]:string
+                  else "";
+    writer.writeln("\"", a[0], "\"", ",", a[1], files);
   }
 }
 
+// Comparator to sort our array representation of the map
+// by the number of occurences of each warning
+record OccurenceComparator {}
+// a and b will be (string, int, int)
+proc OccurenceComparator.compare(a, b) { return b[1] - a[1] ; } // Reverse sort
```

Comment:
I'm not as familiar with writing comparators - would it make sense to explicitly put the type you expect `a` and `b` to have?